src/test/java/org/swisspush/redisques/QueueStatsServiceTest.java

```diff
@@ -0,0 +1,224 @@
+
+        // Verify that async completed correctly
+        testContext.assertTrue(async.count() == 0);
+    }
+
+    /**
+     * Test to verify that dequeue statistics are not collected when the interval is zero or negative.
+     */
+    @Test
+    public void testDequeueStatsNotCalledWhenIntervalIsZeroOrNegative(TestContext testContext) {
+        setUpForDisabledDequeueStats();
+
+        // Mock the configuration to disable dequeue stats
+        Mockito.when(config.isDequeueStatsEnabled()).thenReturn(false);
+
+        // Async setup for the test
+        Async async = testContext.async();
+
+        // Mock the context and mentor
+        Object mockContext = new Object();
+        QueueStatsService.GetQueueStatsMentor<Object> mentor = new QueueStatsService.GetQueueStatsMentor<>() {
+            @Override
+            public boolean includeEmptyQueues(Object ctx) {
+                return true;
+            }
+
+            @Override
+            public int limit(Object ctx) {
+                return 10;
+            }
+
+            @Override
+            public String filter(Object ctx) {
+                return "*";
+            }
+
+            @Override
+            public void onQueueStatistics(java.util.List<QueueStatsService.Queue> queues, Object ctx) {
+                // Ensure that nextDequeueDueTimestampEpochMs is null since stats are not attached
+                testContext.assertTrue(queues.get(0).getNextDequeueDueTimestampEpochMs() == null);
+                async.complete();
+            }
+
+            @Override
+            public void onError(Throwable ex, Object ctx) {
+                testContext.fail(ex);
+            }
+        };
+
+        // Mock fetchQueueNamesAndSize
+        Mockito.doAnswer(invocation -> {
+            BiConsumer<Throwable, QueueStatsService.GetQueueStatsRequest<Object>> callback = invocation.getArgument(1);
+            QueueStatsService.GetQueueStatsRequest<Object> request = invocation.getArgument(0);
+            request.queues = Collections.singletonList(new QueueStatsService.Queue("testQueue", 1));
+            callback.accept(null, request);
+            return null;
+        }).when(queueStatsService).fetchQueueNamesAndSize(Mockito.any(), Mockito.any());
+
+        // Mock fetchRetryDetails
+        Mockito.doAnswer(invocation -> {
+            BiConsumer<Throwable, QueueStatsService.GetQueueStatsRequest<Object>> callback = invocation.getArgument(1);
+            QueueStatsService.GetQueueStatsRequest<Object> request = invocation.getArgument(0);
+            callback.accept(null, request);
+            return null;
+        }).when(queueStatsService).fetchRetryDetails(Mockito.any(), Mockito.any());
+
+        // Call the method getQueueStats
+        queueStatsService.getQueueStats(mockContext, mentor);
+
+        // Verify that attachDequeueStats was NOT called
+        Mockito.verify(queueStatsService, Mockito.never()).attachDequeueStats(Mockito.any(), Mockito.any());
```

Comment:
Do not verify that (internal) methods of `QueueStatsService` have been called.

The `attachDequeueStats` method uses the `DequeueStatisticCollector` which is provided as a dependency in the `QueueStatsService` class. This dependency can be easily mocked. So you don't have to know the inner workings of the `QueueStatsService` class, just verify that the dependency responsible for collecting the statistics is not called.

Reply:
During our call you said that you would implement it in a different location, I moved the validation to the DequeueStatisticCollector, I think it is more appropriate.